File: Exemplos/App_IntentDigitalHUB_Flutter/Elgin_IntentDigitalHUB_Flutter-font/lib/IntentDigitalHubService/SAT/ativar_sat.dart
import 'package:intent_digital_hub/IntentDigitalHubService/SAT/sat_command.dart';

class AtivarSAT extends SatCommand {
  final int _numSessao;
  final int _subComando;
  final String _codAtivacao;
  final String _cnpj;
  final int _cUF;

  AtivarSAT(this._numSessao, this._subComando, this._codAtivacao, this._cnpj,
      this._cUF)
      : super('AtivarSAT');

  @override
  functionParameters() {
    return '"numSessao"'
        ':'
        '$_numSessao'
        ','
        '"subComando"'
        ':'
        '$_subComando'
        ','
        '"codAtivacao"'
        ':'
        '"$_codAtivacao"'
        ','
        '"cnpj"'
        ':'
        '"$_cnpj"'
        ','
        '"cUF"'
        ':'
        '$_cUF';
  }
}

File: Exemplos/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/SAT/AtivarSAT.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService.SAT

import com.elgin.intent_digital_hub.IntentDigitalHubService.SAT.SatCommand

class AtivarSAT(
    private val numSessao: Int,
    private val subComando: Int,
    private val codAtivacao: String,
    private val cnpj: String,
    private val cUF: Int
) : SatCommand("AtivarSAT") {
    override fun functionParameters(): String {
        return "\"numSessao\"" + ":" + numSessao + "," +
                "\"subComando\"" + ":" + subComando + "," +
                "\"codAtivacao\"" + ":" + "\"" + codAtivacao + "\"" + "," +
                "\"cnpj\"" + ":" + "\"" + cnpj + "\"" + "," +
                "\"cUF\"" + ":" + cUF
    }
}